rfcs/0011-removal-managed-zone.md

```diff
@@ -0,0 +1,107 @@
+
+# Motivation
+[motivation]: #motivation
+
+The motiviation for these changes is to simplify the user experience of setting up a DNSPolicy by removing the need for an existing  `ManagedZone` API. As the current reference to the credential is in the ManagedZone API it makes sense to remove that API and move the reference to DNSPolicy.
+
+The key user impact here is to avoid users having to create additional none policy `Kuadrant APIs` in order to be able to leverage the `DNSPolicy` API. 
+
+# Guide-level explanation
+[guide-level-explanation]: #guide-level-explanation
+
+There will no longer be a `ManagedZone` resource that needs to be created as a pre-requisite to setting up a DNSPolicy. Instead it will be expected that the zone already exists in the provider of choice. If it does not exist, you will see this reflected as an error state in the DNSPOlicy status. Credentials will continue to be stored in secrets by default, but these secrets will also be where any additional zone configuration is also applied such as the zone id to target and the root domain in that zone to use.
+
+# Reference-level explanation
+[reference-level-explanation]: #reference-level-explanation
+
+change the DNSPolicy to have local object reference array for provider secrets in the same namespace.
+
+```
+spec:
+  providerRefs
+    - name: myawsroute53
+
+```
+Initially this array will have a limit of 1. The reason for an object array is we expect to want to allow multiple providers secrets to be specified in the future. This allows us to control the number of secrets and allow for possible expansion in the future.
+
+Where we use the existing credential in the kuadrant operator we will now load a list of credentials and iterate through each of them and create a specific DNSRecord instance for the target DNS provider (again initially only 1) populating that DNSRecord with a reference to its required credential. DNSRecord will only have a single reference. The DNS Operator will pick up on these records in the same way it already does except it will load the secret directly rather than via the ManagedZone. The provider type is already specified as part of our existing secret definition:
+
+```
+type: kuadrant.io/aws
+```
+
+ All of the tests will be updated/removed as needed. We will rename the `managedZoneRef` property to be `providerRefs` that will be a required field. This will be an array type with the `maxItems` set to 1 initially. 
+
+
+Any issues updating/contacting a target DNS provider will be reflected in the existing `recordConditions` status via the existing `synchronised` condition.
+
+## Secret structure
+
+under the data field we will expect two new fields to be present
+
+```
+
+ZONE_ID: somezone
+DOMAIN_NAME: *.a.b.com
```

Comment:
The only reason i see needing the DOMAIN_NAME in the secret would be if we intend to look it up, much like we do currently for [ManagedZones](https://github.com/Kuadrant/kuadrant-operator/blob/main/controllers/dns_helper.go#L42), is that the intention? Are we still only creating a DNSRecord if we have a `providerRef` that has a matching domain set? If that is the case, domain name here would not have the wildcard on it, it should match the zone domain name `a.b.com`.

Reply:
Yes I forgot to remove this after our conversation online... Will update. I think we just create the DNSRecord and allow the DNS Opertor to do the work and pass back the result.